app/src/main/java/com/conupods/OutdoorMaps/View/Settings/SettingsPersonalActivity.java

```diff
@@ -26,31 +58,151 @@ protected void onCreate(Bundle savedInstanceState) {
         Button defaultPage = findViewById(R.id.toggle1_1);
         Button infoPage = findViewById(R.id.toggle1_2);
         //Account Options buttons
-        EditText myAccount = findViewById(R.id.email);
-        Button myAccountBox = findViewById(R.id.myAccountBox);
-        Button linkedAccount = findViewById(R.id.linkedAccount);
+        mMyAccount = findViewById(R.id.email);
+        mMyAccountBox = findViewById(R.id.myAccountBox);
+        mLinkedAccount = findViewById(R.id.linkedAccount);
         //Top Menu events
         done.setOnClickListener(view -> startActivityIfNeeded(new Intent(SettingsPersonalActivity.this, MapsActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0));
         defaultPage.setOnClickListener(view -> startActivityIfNeeded(new Intent(SettingsPersonalActivity.this, SettingsActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0));
         infoPage.setOnClickListener(view -> startActivityIfNeeded(new Intent(SettingsPersonalActivity.this, SettingsInfoActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0));
-        //Account options event
-        myAccount.setOnFocusChangeListener((view, hasFocus) -> {
+        //initialise
+        mGoogleCalendarTextView = findViewById(R.id.googleCalendar);
+        //TODO: not sure what this is supposed to do...
+        //my Account options event
+        mMyAccount.setOnFocusChangeListener((view, hasFocus) -> {
             String email = "";
             if (!hasFocus) {
-                email = String.valueOf(myAccount.getText());
-                prefEdit.putString(String.valueOf(myAccount.getId()), email).apply();
-                TextView googleCalendar = findViewById(R.id.googleCalendar);
+                email = String.valueOf(mMyAccount.getText());
+                prefEdit.putString(String.valueOf(mMyAccount.getId()), email).apply();
                 if (!email.equals("")) {
```

Comment:
maybe change the if condition to make sure the email is semi valid. Java String has a endWith() function. maybe make sure the emails ends with @gmail.com

Reply:
That part is actually not my code, im not sure of its purpose in the context of the user settings. @FelixDaCat7 Do we still need this?